qiita_db/metadata_template.py

```diff
@@ -2289,6 +2290,15 @@ def load_template_to_dataframe(fn, strip_whitespace=True):
                                          " cannot be cast into a %s "
                                          "type." % (n, c_dtype))
 
+    # get and clean the required columns
```

Comment:
I think a way to do this is by not writing to a temporary file (see above) and instead write to a StringIO object where you can lowercase all the column names of the first line, look find the location of the columns and then replace only those (leaving the non-required columns untouched). I'm happy to discuss this, cc @adamrp